src/utils/deployments.cairo

```diff
@@ -0,0 +1,84 @@
+mod interface;
+
+use core::pedersen::PedersenTrait;
+use hash::{HashStateTrait, HashStateExTrait};
+use interface::IUniversalDeployer;
+use openzeppelin::utils::serde::SerializedAppend;
+use poseidon::PoseidonTrait;
+use starknet::ClassHash;
+use starknet::ContractAddress;
+
+// 2**251 - 256
+const L2_ADDRESS_UPPER_BOUND: felt252 =
+    0x7ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00;
+const CONTRACT_ADDRESS_PREFIX: felt252 = 'STARKNET_CONTRACT_ADDRESS';
+
+/// Returns the contract address from a `deploy_syscall`.
+/// `deployer_address` should be the zero address if the deployment is origin-independent (deployed from zero).
+/// For more information, see https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-address/
+fn calculate_contract_address_from_deploy_syscall(
+    salt: felt252,
+    class_hash: ClassHash,
+    constructor_calldata: Span<felt252>,
+    deployer_address: ContractAddress
```

Comment:
Wdyt about making this an Option instead of expecting zero? 

If we do this I think we can convert the two udc helpers into just one, by receiving an Option<DeployerInfo>, being DeployerInfo (caller_address, udc_address).

Reply:
Much better solution! Good call